addon/core/handler.ts

```diff
@@ -203,6 +204,30 @@ export default class TableHandler {
     }
   }
 
+  /**
+   * Toggle the selection state of all rows.
+   *
+   * @param {boolean} toggled
+   */
+  toggleSelectAll(toggled: boolean) {
+    this.selection = toggled ? 'all' : [];
+  }
+
+  /**
+   * Add or remove a row to the selected items depending on whether it's already present or not.
+   *
+   * @param {Row} row
+   */
+  updateSelection(row: Row) {
+    this.selection = this.selection instanceof Array ? this.selection : [];
+
+    if (this.selection.includes(row)) {
+      this.selection = this.selection.filter((x: Row) => x !== row);
```

Comment:
You do a lot of `filter` / `includes` for columns / selection etc... you are not afraid by the latency that this could add ?

Reply:
i don't think it will be really noticeable compared to setting a flag like `selected` then having something like `selectedRows` that will do a `filterBy` anyway. Plus, it helps w/ having a single variable for both single selections AND select all.